src/pages/index.js

```diff
@@ -102,45 +192,143 @@ function renderCard(data) {
 }
 
 function getCardElement(cardData) {
-    const card = new Card(cardData, '#card-template', handleImageClick);
+    console.log(cardData);
+    const card = new Card(cardData, '#card-template', handleImageClick, handleCardDelete, handleCardLike, handleCardDislike);
     return card.getView();
 }
 
-function handleImageClick(cardData) {
-    popupWithImage.open(cardData);
+function showErrorMessage(err){
+    const messageElement = document.querySelector('.modal__button');
+    messageElement.textContent = `Error: ${err}`;
 }
 
 /* -------------------------------------------------------------------------- */
 /*                               Event Handlers                               */
 /* -------------------------------------------------------------------------- */
 
+function handleImageClick(cardData) {
+    popupWithImage.open(cardData);
+}
+
+function handleCardLike(card){
+    api
+        .likeCard(card._id)
+        .then(() => {
+            card.cardIsLiked(true);
+        })
+        .catch((err) => {
+            console.log(err);
+        });
+}
+
+function handleCardDislike(card) {
+    api
+        .dislikeCard(card._id)
+        .then(() => {
+            card.cardIsLiked(false);
+        })
+        .catch((err) => {
+            console.error(err);
+        });
+}
+
+function handleSubmit(request, popupInstance, loadingText = 'Saving...'){
+    popupInstance.renderLoading(true, loadingText);
+    request()
+        .then(() => {
+            popupInstance.close();
+        })
+        .catch(console.error)
+        .finally(() => {
+            popupInstance.renderLoading(false);
+        });
+}
+
 function handleProfileEditSumbit (userData) {
-    console.log(userData);
     const name = userData.name;
     const job = userData.bio;
-    userInfo.setUserInfo(name, job);
-    popupEditForm.close();
+    console.log(name);
+    console.log(job);
+    function makeRequest(){
+        return api.updateProfileInfo(name, job)
+        /*.then(() => {
+            return api.getUserInfo();
+        })*/
+        .then(() => {
+            userInfo.setUserInfo(name, job);
+            popupEditForm.close();
+            profileEditFormElement.reset();
+        })
+        .catch((err) => {
```

Comment:
`catch` is not needed if you use `handleSubmit`. Please, add `showErrorMessage();` for the `catch` block in line 241 if you want to call it